tx_extension_clip/matcher.py

```diff
@@ -7,11 +7,70 @@
 
 from tx_extension_clip.config import BITS_IN_CLIP
 from tx_extension_clip.utils.uint import int64_to_uint64, uint64_to_int64
+from tx_extension_clip.utils.distance import similarity_to_distance
 
 CLIP_HASH_TYPE = t.Union[str, bytes]
 
 
+def _to_python_int(value: t.Any) -> int:
+    """Convert numpy integer types to Python int (for Hamming distances)."""
+    if hasattr(value, "item"):
+        return int(value.item())
+    return int(value)
+
+
+def _to_python_float(value: t.Any) -> float:
+    """Convert numpy float types to Python float (for cosine similarity/distance)."""
+    if hasattr(value, "item"):
+        return float(value.item())
+    return float(value)
+
+
+class CLIPFloatIndex(ABC):
```

Comment:
It confuses me that this class has the same name as the one in tx_extension_clip/index.py